lib/import_export/bloc/import_export_bloc.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:path_provider/path_provider.dart';
+import 'dart:io';
+import 'package:intl/intl.dart';
+import 'package:share_plus/share_plus.dart';
+
+import 'package:gibsonify_api/gibsonify_api.dart';
+import 'package:gibsonify_repository/gibsonify_repository.dart';
+
+part 'import_export_event.dart';
+part 'import_export_state.dart';
+
+class ImportExportBloc extends Bloc<ImportExportEvent, ImportExportState> {
+  final GibsonifyRepository _gibsonifyRepository;
+
+  ImportExportBloc({required GibsonifyRepository gibsonifyRepository})
+      : _gibsonifyRepository = gibsonifyRepository,
+        super(const ImportExportState()) {
+    on<DataSavedToFiles>(_onDataSavedToFiles);
+    on<DataShared>(_onDataShared);
+  }
+
+  void _onDataSavedToFiles(
```

Comment:
Looks good for now, in the future we should probably separate exporting recipe data and collection data as two separate options, same for saving to internal memory etc

Reply:
Yup, I was thinking that we should also have options to export all Collections, not just the finished ones, but I wanted to stick to current feature requests. Currently, exporting everything together is the most straightforward (and hopefully also the least-bug-ridden) option, for example synchronizing the files to have the same timestamp in the name if they were saved asynchronously (and possibly by different blocs) would just be more cumbersome. I've done a couple of designs of the export while making this PR and for now, having everything bundled together in just one bloc a few event handlers seems the most straightforward, with possibility of extending and atomizing the functionality in the future.